be/src/exec/data_sink.cpp

```diff
@@ -396,63 +404,108 @@ Status DataSink::decompose_data_sink_to_pipeline(pipeline::PipelineBuilderContex
         TableDescriptor* table_desc =
```

Comment:
Better to create a new static function in IcebergTableSink to move all these logics into it.

Reply:
not a good idea to implement all this internal logic here.

Reply:
done. Moved detailed logic into subclasses.